src/pip/_vendor/colorama/ansitowin32.py

```diff
@@ -136,12 +137,25 @@ def get_win32_calls(self):
             }
         return dict()
 
+    def _flush(self):
```

Comment:
You've modified a vendored library - colorama. What's the reason you had to do this?

pip actively avoids patching it's vendored libraries in favour of making changes upstream and absorbing those changes when a new release is made.

Reply:
As explained it also in conversation, patching logger is not enough.  

In windows, as soon as you patch it and try to redirect to head, that the next point where it fails on broken-pipe.
And not only it simply fails, but if you ignore also at that point pipe-errors, you then get a mysterious EINVAL OSError:
```
   File "pip\_vendor\colorama\ansitowin32.py", line 143, in _flush
   self.wrapped.flush()
   OSError: [Errno 22] Invalid argument
```
If you ignore also that, then it works.

Now i'm also surprised that Colorama has not fixed it, and i search a bit and did not fimnd anything related.
Certainly if it is fixable, it should be fixed there.